### DISCUSSION TITLE
port 65339 for ULA+ noread?

### DISCUSSION THREAD

**Discussion by tuFFo126** [UzixLS/zx-sizif-512#45] [General]
According to the ULA+ specification, port 65339 is read/write. In sizif512 it looks like the port for registers (0-63) is write-only, it can only read from register 64 and only bit0.
I also have Just Speccy 128k with SLAM+128 and it matches the specifications there.
![20231006_141951](https://github.com/UzixLS/zx-sizif-512/assets/124886433/f437559f-e8ba-44e5-b1ae-309fdb6792c3)
![20231006_142429](https://github.com/UzixLS/zx-sizif-512/assets/124886433/6d803276-c835-4808-8d46-5a33c5d5bce3)
![20231006_142014](https://github.com/UzixLS/zx-sizif-512/assets/124886433/c531e02d-a34e-424f-b0d2-9a9d1db3c807)

**Comment by UzixLS** [comment]
Reading isn't implemented because Sizif does not have enough cells inside CPLD to fit all ULA+ registers and instead use SRAM to store em. Reading would make things a much more complex.
Does it really required by some software or just for show?

**Reply by tuFFo126** [reply]
I understand correctly, what about false colors in ULA+ in the previous query?

**Comment by UzixLS** [comment]
Implemented with commit 8e092541dd402329574b6fbecd31f13e869f7e80
Now all ULA+ registers can be read.
